### DISCUSSION TITLE
Is there a way to invoke customized ErrorHandler for earlyRecordInterceptor process in spring-kafka 2.8

### DISCUSSION THREAD

**Discussion by LuckyChristy** [spring-projects/spring-kafka#2080] [Q&A]
Hi guys, we are upgrade spring-kafka from 2.5x to 2.8, we find in 2.8, you put the execution process of `RecordInterceptor` into `earlyRecordInterceptor`, but during the `earlyRecordInterceptor` execution, there is no ErrorHandler can be invoked. 

in my case (non-transactional), i set a customized interceptor to kafkaListener like below:
```
@Bean
    public ConcurrentKafkaListenerContainerFactory<Object, Object> concurrentKafkaListenerContainerFactory(ConcurrentKafkaListenerContainerFactory<Object, Object> concurrentKafkaListenerContainerFactory, MyConsumerInterceptor<Object, Object> myConsumerInterceptor, DeadLetterPublishingRecoverer deadLetterPublishingRecoverer) {
        concurrentKafkaListenerContainerFactory.setRecordInterceptor(myConsumerInterceptor);  // set my customized Interceptor
        concurrentKafkaListenerContainerFactory.getContainerProperties().setAckMode(ContainerProperties.AckMode.RECORD);
        concurrentKafkaListenerContainerFactory.setErrorHandler(new SeekToCurrentErrorHandler(deadLetterPublishingRecoverer, new FixedBackOff(0, 0))); // my errorHandler
        return concurrentKafkaListenerContainerFactory;
    }
```
If there is an exception occur in my customized `RecordInterceptor`, I want it can be handled by my errorHandler `SeekToCurrentErrorHandler`. Spring-kafka2.5x, 2.6x can do that, but spring-kafka 2.8 cannot. I already did an investigation, its due to you invoke my recordInterceptor in method `checkEarlyIntercept` (no Errorhandler invoked if there is an exception, [code](https://github.com/spring-projects/spring-kafka/blob/2.6.x/spring-kafka/src/main/java/org/springframework/kafka/listener/KafkaMessageListenerContainer.java#L1991)), not in `doInvokeRecordListener` method (invoked errorHandler there is an excetiption, [code](https://github.com/spring-projects/spring-kafka/blob/2.6.x/spring-kafka/src/main/java/org/springframework/kafka/listener/KafkaMessageListenerContainer.java#L2034)). 

I already list the question in stackoverflow [here](https://stackoverflow.com/questions/70798263/is-there-a-way-to-invoke-customized-errorhandler-for-earlyrecordinterceptor-in-s), do you have any idea how to invoke errorhandler for `earlyRecordInterceptor`?

Thank you~

**Comment by garyrussell** [comment]
Don't ask questions in multiple places; it is a waste of your time and ours.

I already answered there https://stackoverflow.com/questions/70798263/is-there-a-way-to-invoke-customized-errorhandler-for-earlyrecordinterceptor-in-s/70835382#70835382
